sdk/include/opentelemetry/sdk/trace/span_data.h

```diff
@@ -76,6 +86,11 @@ class SpanData final : public Recordable
     parent_span_id_ = parent_span_id;
   }
 
+  void SetAttribute(nostd::string_view key, const common::AttributeValue &&value) noexcept override
+  {
+    attributes_[std::string(key)] = value;
+  }
+
   void AddEvent(nostd::string_view name, core::SystemTimestamp timestamp) noexcept override
```

Comment:
How do you plan on storing events on Recordable?

Reply:
I think we'll need to come up with a proper `struct` type encapsulating an event, but I didn't spend much time thinking about this yet.